File: app/src/main/java/com/jeeps/gamecollector/model/GameHoursStats.java
package com.jeeps.gamecollector.model;

import com.jeeps.gamecollector.model.hltb.GameplayHoursStats;

import java.io.Serializable;

public class GameHoursStats implements Serializable {
    private double gameplayMain;
    private double gameplayMainExtra;
    private double gameplayCompletionist;

    public GameHoursStats(double gameplayMain, double gameplayMainExtra, double gameplayCompletionist) {
        this.gameplayMain = gameplayMain;
        this.gameplayMainExtra = gameplayMainExtra;
        this.gameplayCompletionist = gameplayCompletionist;
    }

    public GameHoursStats(GameplayHoursStats gameplayHoursStats) {
        this.gameplayMain = gameplayHoursStats.getGameplayMain();
        this.gameplayMainExtra = gameplayHoursStats.getGameplayMainExtra();
        this.gameplayCompletionist = gameplayHoursStats.getGameplayCompletionist();
    }

    public GameHoursStats() {
    }

    public double getGameplayMain() {
        return gameplayMain;
    }

    public void setGameplayMain(double gameplayMain) {
        this.gameplayMain = gameplayMain;
    }

    public double getGameplayMainExtra() {
        return gameplayMainExtra;
    }

    public void setGameplayMainExtra(double gameplayMainExtra) {
        this.gameplayMainExtra = gameplayMainExtra;
    }

    public double getGameplayCompletionist() {
        return gameplayCompletionist;
    }

    public void setGameplayCompletionist(double gameplayCompletionist) {
        this.gameplayCompletionist = gameplayCompletionist;
    }
}

File: app/src/main/java/com/jeeps/gamecollector/remaster/data/model/data/games/GameHoursStats.kt
package com.jeeps.gamecollector.remaster.data.model.data.games

import com.jeeps.gamecollector.remaster.data.model.data.hltb.GameplayHoursStats
import java.io.Serializable

data class GameHoursStats(
    var gameplayMain: Double = 0.0,
    var gameplayMainExtra: Double = 0.0,
    var gameplayCompletionist: Double = 0.0
) : Serializable {
    constructor(gameplayHoursStats: GameplayHoursStats) : this(
        gameplayHoursStats.gameplayMain,
        gameplayHoursStats.gameplayMainExtra,
        gameplayHoursStats.gameplayCompletionist)

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as GameHoursStats

        if (gameplayMain != other.gameplayMain) return false
        if (gameplayMainExtra != other.gameplayMainExtra) return false
        if (gameplayCompletionist != other.gameplayCompletionist) return false

        return true
    }

    override fun hashCode(): Int {
        var result = gameplayMain.hashCode()
        result = 31 * result + gameplayMainExtra.hashCode()
        result = 31 * result + gameplayCompletionist.hashCode()
        return result
    }

}